### PR TITLE
Fix/5267

### LINKED ISSUE #5267: [Network] node's chain tip can get delayed for a while
I've noticed that once a node has sync'ed that if it fails to keep pace with block push, it's not always able to download missing blocks in very recent tenures.  This is a big problem, because it'll mean that an afflicted node will be one or more Bitcoin blocks behind the chain tip.  I have a few hunches as to the root cause, but I'm not certain of anything yet.

### PR DESCRIPTION
**Pull request by jcnelson**
Fixes #5267 by making it so that an idle downloader can be assigned to a different peer.

### CODE REVIEW

**Review by obycode — APPROVED** [review]
Nice! LGTM

File: stackslib/src/net/download/nakamoto/download_state_machine.rs [hunk 1]
```diff
@@ -983,9 +983,9 @@ impl NakamotoDownloadStateMachine {
             prev_schedule
         };
 
-        debug!("new schedule: {:?}", schedule);
-        debug!("new available: {:?}", &available);
-        debug!("new tenure_block_ids: {:?}", &tenure_block_ids);
+        test_debug!("new schedule: {:?}", schedule);
+        test_debug!("new available: {:?}", &available);
+        test_debug!("new tenure_block_ids: {:?}", &tenure_block_ids);
```
Comment (kantai) [inline stackslib/src/net/download/nakamoto/download_state_machine.rs:988]: Can you consolidate into a single `test_debug!` line?

**Review by kantai — DISMISSED** [review]

File: stackslib/src/net/download/nakamoto/tenure_downloader.rs [hunk 2]
```diff
@@ -558,10 +581,11 @@ impl NakamotoTenureDownloader {
         response: StacksHttpResponse,
     ) -> Result<Option<Vec<NakamotoBlock>>, NetError> {
         let handle_result = match self.state {
-            NakamotoTenureDownloadState::GetTenureStartBlock(_block_id) => {
+            NakamotoTenureDownloadState::GetTenureStartBlock(_block_id, _start_request_time) => {
```
Comment (hstove) [inline stackslib/src/net/download/nakamoto/tenure_downloader.rs:584]: Nit here (and a few other similar lines) - don't prefix with `_` if it gets used

**Review by hstove — CHANGES_REQUESTED** [review]
Pretty much LGTM, just a note about the `_` prefix. You also have some compilation errors in `net/tests/downloader/nakamoto` if you haven't seen that yet

**Review by hstove — APPROVED** [review]

**Review by obycode — APPROVED** [review]

**Pull request merged**